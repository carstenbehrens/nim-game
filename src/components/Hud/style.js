import styled from 'styled-components'

export const HudContainer = styled.div`
	height: 30%;
	width: 100%;
	color: white;
	font-size: 2rem;
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: space-around;
`;

export const FinishTurn = styled.button`
	height: 4rem;
	width: 8rem;
	color: white;
	background: none;
	font-weight: bold;
	cursor: pointer;
	text-transform: uppercase;
`;