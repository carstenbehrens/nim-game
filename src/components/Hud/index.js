import React from 'react';
import { HudContainer, FinishTurn} from './style'

export default ({ userIsCurrentPlayer, handleFinishTurn, matches }) => {
	return (
		<HudContainer>
			<FinishTurn onClick={() => handleFinishTurn(matches)}>
				Finish Turn
			</FinishTurn>
			<p>Player: </p>
			{userIsCurrentPlayer ? 'User' : 'Computer'}
		</HudContainer>
	);
};
