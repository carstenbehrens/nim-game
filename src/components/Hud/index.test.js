import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json'
import Hud from './index';

describe('Hud', () => {
  it('renders without crashing', () => {
    shallow(<Hud />);
  });

  it('matches snapshot', () => {
    const wrapper = mount(<Hud userIsCurrentPlayer={true}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})