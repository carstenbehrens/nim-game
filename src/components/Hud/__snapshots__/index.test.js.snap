// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hud matches snapshot 1`] = `
<_default
  userIsCurrentPlayer={true}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "hLJrHI",
            "rules": Array [
              "
  height: 20%;
	width: 100%;
	color: white;
	font-size: 2rem;
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: space-around;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa hLJrHI"
      >
        <styled.button
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "kcoHmH",
                  "rules": Array [
                    "
	height: 4rem;
	width: 8rem;
	color: white;
	background: none;
	font-weight: bold;
	cursor: pointer;
	text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "button",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="sc-bwzfXH kcoHmH"
              onClick={[Function]}
            >
              Finish Turn
            </button>
          </StyledComponent>
        </styled.button>
        <p>
          Player: 
        </p>
        User
      </div>
    </StyledComponent>
  </styled.div>
</_default>
`;
